### PR TITLE
Allow whitespace characters within component regex

### PR DESCRIPTION
**Pull request by Archmonger**
## Description

I realized some handlebars formatters will add indentation and newlines. Therefore, I've made the component regex a bit more robust to handle those situations.

## Checklist:

Please update this checklist as you complete each item:

-   [X] Tests have been included for all bug fixes or added functionality.
-   [x] The `changelog.rst` has been updated with any significant changes, if necessary.
-   [X] GitHub Issues which may be closed by this PR have been linked.

### CODE REVIEW

File: src/django_idom/utils.py [hunk 1]
```diff
@@ -13,8 +13,19 @@
 from django_idom.config import IDOM_REGISTERED_COMPONENTS
 
 
-COMPONENT_REGEX = re.compile(r"{% *component +((\"[^\"']*\")|('[^\"']*'))(.*?)%}")
 _logger = logging.getLogger(__name__)
+_component_tag = r"component"
+_component_path = r"((\"[^\"']*\")|('[^\"']*'))"
+_component_kwargs = r"((.*?|\s*?)*)"
```
Comment (rmorshea) [inline src/django_idom/utils.py:19]: While we're at it. Can we add names to these groups with `(?P<...>)`? Will just help out readability when matching.

**Comment by rmorshea** [comment]
In usages of the regex pattern you can use `pattern.match(...).group(group_name)` instead of grabbing them by index.

**Comment by Archmonger** [comment]
`pattern.match(...).group(group_name)` only works with single objects. To "find all" I needed to use `finditer` and `groupdict`.

I don't believe I'm using the index anywhere anymore.

File: src/django_idom/utils.py [hunk 2]
```diff
@@ -102,11 +113,14 @@ def _get_components(self, templates: set[str]) -> set[str]:
         for template in templates:
             with contextlib.suppress(Exception):
                 with open(template, "r", encoding="utf-8") as template_file:
-                    match = COMPONENT_REGEX.findall(template_file.read())
-                    if not match:
+                    regex_iterable = COMPONENT_REGEX.finditer(template_file.read())
+                    if not regex_iterable:
                         continue
```
Comment (rmorshea) [inline src/django_idom/utils.py:118]: This is always going to be true. If the iterable is empty `update()` will do nothing anyway.
```suggestion
```

Comment (rmorshea) [inline src/django_idom/utils.py:118]: No need for this check. The `update()` will do nothing if the iterable is empty
```suggestion
```

**Review by rmorshea — CHANGES_REQUESTED** [review]

File: src/django_idom/utils.py [hunk 3]
```diff
@@ -102,11 +113,14 @@ def _get_components(self, templates: set[str]) -> set[str]:
...
                     components.update(
-                        [group[0].replace('"', "").replace("'", "") for group in match]
+                        [
+                            match.groupdict()["path"].replace('"', "").replace("'", "")
```
Comment (rmorshea) [inline src/django_idom/utils.py:121]: I'm pretty sure you can just do this if you only need one named group:
```suggestion
                            match.group("path").replace('"', "").replace("'", "")
```

**Review by rmorshea — APPROVED** [review]

**Pull request merged**